repos/system_upgrade/common/libraries/overlaygen.py

```diff
@@ -206,15 +431,59 @@ def _mount_dnf_cache(overlay_target):
 
 
 @contextlib.contextmanager
-def create_source_overlay(mounts_dir, scratch_dir, xfs_info, storage_info, mount_target=None):
+def create_source_overlay(mounts_dir, scratch_dir, xfs_info, storage_info, mount_target=None, scratch_reserve=0):
     """
     Context manager that prepares the source system overlay and yields the mount.
+
+    The in-place upgrade itself requires to do some changes on the system to be
+    able to perform the in-place upgrade itself - or even to be able to evaluate
+    if the system is possible to upgrade. However, we do not want to (and must not)
+    change the original system until we pass beyond the point of not return.
+
+    For that purposes we have to create a layer above the real host file system,
+    where we can safely perform all operations without affecting the system
+    setup, rpm database, etc. Currently overlay (OVL) technology showed it is
+    capable to handle our requirements good enough - with some limitations.
+
+    This function prepares a disk image and an overlay layer for each
+    mountpoint configured in /etc/fstab, excluding those with FS type noted
+    in the OVERLAY_DO_NOT_MOUNT set. Such prepared OVL images are then composed
+    together to reflect the real host filesystem. In the end everything is cleaned.
+
+    The new solution can be now problematic for system with too many partitions
+    and loop devices. For such systems we keep for now the possibility of the
+    fallback to an old solution, which has however number of issues that are
+    fixed by the new design. To fallback to the old solution, set envar:
+        LEAPP_OVL_FALLBACK=1
```

Comment:
That's true. But we expect the use of that variable by user when they realize the new solution does not work. Just it is not executed automatically in case of problems. Any suggestion for the name of the envar? Adding some cadidates below:

* LEAPP_OVL_FALLBACK
* LEAPP_OVL_OLD_SOLUTION
* LEAPP_OVL_LEGACY
* ??

Reply:
I am with @MichalHe here - fallback implies automatic action taken upon another one failing.
`LEAPP_OVL_LEGACY` is the best one from the list imho.

Reply:
I think that with the context you provided both FALLBACK and LEGACY make sense.

Reply:
I've spent some time with thesaurus looking for good-sounding words that would express the purpose of the env var, but I have found none. Let's go with `LEGACY ` as that is something already used to express something very similar in the field, e.g., GRUB2 vs GRUB legacy.

Reply:
+1 to LEAPP_OVL_LEGACY

Reply:
LEAPP_OVL_LEGACY sounds the best to me also.

Reply:
So LEAPP_OVL_LEGACY be it! thanks!

Reply:
done